nifi-frontend/src/main/frontend/apps/nifi-registry/src/app/pages/expolorer/feature/ui/import-new-flow-version-dialog/import-new-flow-version-dialog.component.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ElementRef, Inject, ViewChild } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MAT_DIALOG_DATA, MatDialogModule } from '@angular/material/dialog';
+import { CloseOnEscapeDialog } from '@nifi/shared';
+import { FormBuilder, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { importNewFlow } from 'apps/nifi-registry/src/app/state/droplets/droplets.actions';
+import { Droplets } from 'apps/nifi-registry/src/app/state/droplets';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatSelectModule } from '@angular/material/select';
+import { MatInputModule } from '@angular/material/input';
+import { MatButtonModule } from '@angular/material/button';
+
+interface Data {
```

Comment:
IMO it is worh it to export the interface and use in 
    
    this.dialog.open<ImportNewFlowVersionDialogComponent, ImportNewFlowVersionDialogData>(ImportNewFlowVersionDialogComponent, { 
        data // becomes type safe
    }

```suggestion
export interface ImportNewFlowVersionDialogData {
```